src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -694,6 +695,20 @@ private static void WriteDataSources(DirectoryWriter dir, CanvasDocument app, Er
 
                 dir.WriteAllJson(DataSourcesDir, new FilePath(filename), dataSourceStateToWrite);
             }
+
+            if (app._dataSourceReferences != null)
+            {
+                foreach (var kvp in app._dataSourceReferences)
+                {
+                    if (kvp.Value?.components?.Any() == true)
```

Comment:
>if (kvp.Value?.components?.Any() == true) [](http://example.com/codeflow?start=20&length=41)

This may be too specific - 
can we just make this a catch-all (ie, regardless of whether it's a component, anything not captured by above L682-L691 gets written here)